### PR TITLE
Feature/get submission invitations

### PR DESCRIPTION
**Pull request by mohituniyal**
Please review

Refer PR #127

### CODE REVIEW

File: openreview/tools.py [hunk 1]
```diff
@@ -703,3 +698,39 @@ def post_submission_groups(client, conference_id, submission_invite, chairs):
...
+        
+        for g in groups :
+            
+            #Calculate the epoch for current timestamp
+            now = int(time.time()*1000)
+
+            # Assume the invitation is closed unless proven open
+            due = 0
+
+            if g.to_json()['duedate'] != None and now < g.to_json()['duedate']:
```
Comment (melisabok) [inline openreview/tools.py]: no need to convert to json to get the duedate

File: openreview/tools.py [hunk 2]
```diff
@@ -703,3 +698,39 @@ def post_submission_groups(client, conference_id, submission_invite, chairs):
...
+                due = 1
+            
+            if status == "open" and due == 1:
```
Comment (melisabok) [inline openreview/tools.py]: @mspector I need your opinion about how to filter the invitations.

Comment (melisabok) [reply openreview/tools.py]: I think we don't need to filter the result locally, we can add the filter to the get_invitations call.

Comment (mspector) [reply openreview/tools.py]: do you think it should be filtered in the backend, or in the python library?

Comment (melisabok) [reply openreview/tools.py]: the homepage of openreview gets the open submissions using these parameters:

```
regex: .*/-/[Ss]ubmission
minduedate: 1531505712680
```

I'm not sure if minduedate is implemented in the library.

File: openreview/tools.py [hunk 3]
```diff
@@ -703,3 +698,39 @@ def post_submission_groups(client, conference_id, submission_invite, chairs):
...
+    '''
+    Returns a list of invitations ids according to the status.
+    
+    :arg client: Object of :class:`~openreview.Client` class
+    :arg status: Status filter for invitations. Default value is "all". This is a string with following possible value: all, open and closed.
+    '''
+    
+    try:
+        # Get a list of all group with invitations satisfying the regex
+        groups = client.get_invitations(regex='.*/-/.*submission')
```
[the remark this answers is not in the document]
Comment (mohituniyal) [reply openreview/tools.py]: Corrected

File: openreview/tools.py (hunk 1 above)
Comment (mohituniyal) [reply openreview/tools.py]: Corrected

File: openreview/tools.py [hunk 4]
```diff
@@ -703,3 +698,39 @@ def post_submission_groups(client, conference_id, submission_invite, chairs):
...
+    '''
+    Returns a list of invitations ids according to the status.
+    
+    :arg client: Object of :class:`~openreview.Client` class
+    :arg status: Status filter for invitations. Default value is "all". This is a string with following possible value: all, open and closed.
+    '''
+    
+    try:
+        # Get a list of all invitations satisfying the regex
+        invitations = client.get_invitations(regex='.*/-/.*[sS]ubmission.*')
```
Comment (melisabok) [inline openreview/tools.py]: We can filter the invitation in the server using the parameters: minduedate.
 
In this case the status parameter can be renamed to open_only: [True, False]

if open_only is true then send minduedate = now
if open_only is false or None then don't send minduedate parameter. 

Does it make sense?

Comment (mohituniyal) [reply openreview/tools.py]: Suggested changes have been implemented.

File: openreview/tools.py [hunk 5]
```diff
@@ -703,6 +698,37 @@ def post_submission_groups(client, conference_id, submission_invite, chairs):
...
+            invitations = client.get_invitations(regex='.*/-/.*[sS]ubmission.*')
+        
+        # For each group in the list, append the invitation id to a list
+        invitation_ids = []
+        
+        for inv in invitations :
+            invitation_ids.append(inv.id)
+        
+        return invitation_ids
+    except openreview.OpenReviewException as e:
```
Comment (melisabok) [inline openreview/tools.py]: I would follow the same rules that get_all_venues and don't catch the exception

File: openreview/tools.py [hunk 6]
```diff
@@ -703,6 +698,37 @@ def post_submission_groups(client, conference_id, submission_invite, chairs):
...
+    Example Usage:
+    >>> get_submission_invitations(c,True)
+    [u'machineintelligence.cc/MIC/2018/Conference/-/Submission', u'machineintelligence.cc/MIC/2018/Abstract/-/Submission', u'ISMIR.net/2018/WoRMS/-/Submission', u'OpenReview.net/Anonymous_Preprint/-/Submission']
+    '''
+    
+    try:
+        #Calculate the epoch for current timestamp
+        now = int(time.time()*1000)
+        
+        if open_only == True:
```
Comment (melisabok) [inline openreview/tools.py]: I would set the minduedate value in the if and do the call get_invitation once so we reduced the repeated code:

```
duedate = now if open_only else None
invitations = client.get_invitations(regex='.*/-/.*[sS]ubmission.*',minduedate=duedate)

```

File: openreview/tools.py [hunk 7]
```diff
@@ -703,6 +698,37 @@ def post_submission_groups(client, conference_id, submission_invite, chairs):
...
+        
+        if open_only == True:    
+            invitations = client.get_invitations(regex='.*/-/.*[sS]ubmission.*',minduedate=now)
+        else:
+            invitations = client.get_invitations(regex='.*/-/.*[sS]ubmission.*')
+        
+        # For each group in the list, append the invitation id to a list
+        invitation_ids = []
+        
+        for inv in invitations :
```
Comment (melisabok) [inline openreview/tools.py]: There is a nice way to iterate a list in a single line with Python called [for comprehension](https://docs.python.org/2/tutorial/datastructures.html)

```
invitation_ids = [i.id for i un invitations]
```

Comment (mohituniyal) [reply openreview/tools.py]: All 3 changes have been applied.

**Pull request merged**